File: java/CyberFactSoftwareVulnerabilities.java
import javax.ws.rs.client.Client;
import javax.ws.rs.client.ClientBuilder;
import javax.ws.rs.client.WebTarget;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.MultivaluedHashMap;
import javax.ws.rs.core.MultivaluedMap;

public class CyberFactSoftwareVulnerabilities {
    
    private static final String API_BASE_URL = "https://www.surfwatchlabs.com:443/api/v3";

    public static void main( String[] args ) throws Exception {

        Client restClient = ClientBuilder.newClient();
        WebTarget target = restClient.target( API_BASE_URL )
                .path( "/cyberFacts/softwareVulnerabilities" )
                .queryParam( "startDate", "2015-06-25" )
                .queryParam( "endDate", "2015-06-25" );

        MultivaluedMap headers = new MultivaluedHashMap<>();
        headers.add( "app_id", "your_app_id" );
        headers.add( "app_key", "your_app_key" );

        String response = target
                .request( MediaType.APPLICATION_JSON )  // alternately set "Content-Type" header
                .headers( headers )
                .get( String.class );

        System.out.print( response );
    }
    
}

File: ruby/cyberfact_software_vulnerabilities.rb
require 'rest_client'
require 'json'

url = 'https://www.surfwatchlabs.com:443/api/v3/cyberFacts/softwareVulnerabilities'

header = {
  'content_type' => 'application/json',
  'app_key' => ENV['SURFWATCH_ANALYTICS_APP_KEY'],
  'app_id' => ENV['SURFWATCH_ANALYTICS_APP_ID'],
  params: {
    'startDate' => '2015-06-25',
    'endDate' => '2015-06-25'
  }
}
response = RestClient.get(url, header)
results = JSON.parse(response)
puts results